src/Features/CSharp/Portable/QuickInfo/CSharpDiagnosticAnalyzerQuickInfoProvider.cs

```diff
@@ -0,0 +1,108 @@
+{
+    [ExportQuickInfoProvider(QuickInfoProviderNames.DiagnosticAnalyzer, LanguageNames.CSharp), Shared]
+    [ExtensionOrder(After = QuickInfoProviderNames.Syntactic)]
+    internal class CSharpDiagnosticAnalyzerQuickInfoProvider : CommonQuickInfoProvider
+    {
+        private readonly IDiagnosticAnalyzerService _diagnosticAnalyzerService;
+
+        [ImportingConstructor]
+        [SuppressMessage("RoslynDiagnosticsReliability", "RS0033:Importing constructor should be [Obsolete]", Justification = "Used in test code: https://github.com/dotnet/roslyn/issues/42814")]
+        public CSharpDiagnosticAnalyzerQuickInfoProvider(IDiagnosticAnalyzerService diagnosticAnalyzerService)
+        {
+            _diagnosticAnalyzerService = diagnosticAnalyzerService;
+        }
+
+        protected override async Task<QuickInfoItem?> BuildQuickInfoAsync(
+            Document document,
+            SyntaxToken token,
+            CancellationToken cancellationToken)
+        {
+            var errorCode = token.Parent switch
+            {
+                PragmaWarningDirectiveTriviaSyntax directive
+                    => token.IsKind(SyntaxKind.EndOfDirectiveToken)
+                        ? directive.ErrorCodes.LastOrDefault() as IdentifierNameSyntax
+                        : directive.ErrorCodes.FirstOrDefault() as IdentifierNameSyntax,
+                IdentifierNameSyntax { Parent: PragmaWarningDirectiveTriviaSyntax _ } identifier
+                    => identifier,
+                _ => null,
+            };
+
+            if (errorCode != null)
+            {
+                return GetQuickInfoFromSupportedDiagnosticsOfProjectAnalyzers(document, errorCode);
+            }
+
+            return null;
+        }
+
+        private QuickInfoItem? GetQuickInfoFromSupportedDiagnosticsOfProjectAnalyzers(Document document,
+            IdentifierNameSyntax errorCode)
+        {
+            var infoCache = _diagnosticAnalyzerService.AnalyzerInfoCache;
+            var hostAnalyzers = document.Project.Solution.State.Analyzers;
+            var groupedDiagnostics = hostAnalyzers.GetDiagnosticDescriptorsPerReference(infoCache, document.Project).Values;
+            var supportedDiagnostics = groupedDiagnostics.SelectMany(d => d);
+            var diagnosticDescriptor = supportedDiagnostics.FirstOrDefault(d => d.Id == errorCode.Identifier.ValueText);
+            if (diagnosticDescriptor != null)
+            {
+                var description =
+                    diagnosticDescriptor.Title.ToStringOrNull() ??
+                    diagnosticDescriptor.Description.ToStringOrNull() ??
+                    diagnosticDescriptor.MessageFormat.ToStringOrNull() ??
+                    diagnosticDescriptor.Id;
+
+                return CreateQuickInfo(errorCode, description);
+            }
+
+            return null;
+        }
+
+        private static QuickInfoItem CreateQuickInfo(IdentifierNameSyntax pragmaWarningDiagnosticId, string description,
```

Comment:
Instead of quickinfo just having the description, can we match it to the quickinfo we generate when hovering over the squiggles, i.e. have the diagnostic ID with a hyperlink to the help link in the descriptor, followed by the colon and the title? @sharwell Can you point to the provider you authored to show this info? Likely that we can share all the code that goes from diagnostic descriptor to the quick info item between both these quick info providers.

Reply:
Seems like the logic is here: http://sourceroslyn.io/#Microsoft.CodeAnalysis.EditorFeatures/Implementation/Diagnostics/AbstractDiagnosticsAdornmentTaggerProvider.cs,55. We directly associate it with the diagnostic tag created by the tagger provider.

Reply:
Done in 7ba781c. I think I came very close to what is done in the `AbstractDiagnosticsAdornmentTaggerProvider`.